src/Interpreters/HashJoin.cpp

```diff
@@ -749,8 +750,25 @@ class AddedColumns
         if constexpr (has_defaults)
             applyLazyDefaults();
 
-        for (size_t j = 0, size = right_indexes.size(); j < size; ++j)
-            columns[j]->insertFrom(*block.getByPosition(right_indexes[j]).column, row_num);
+        if (is_join_get)
```

Comment:
Maybe it can be `constexpr` like `has_defaults`, but I suppose it won't significantly affect performance